src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -343,12 +342,50 @@ public void Initialize()
             Task.Run(ClusterMaintenanceSetupTask, ServerShutdown);
         }
 
-        private void OnStateChanged(object sender, RachisConsensus.State e)
+        private void OnStateChanged(object sender, RachisConsensus.State state)
         {
             using (ContextPool.AllocateOperationContext(out TransactionOperationContext context))
             using (context.OpenReadTransaction())
             {
                 NotificationCenter.Add(ClusterTopologyChanged.Create(GetClusterTopology(context), LeaderTag, NodeTag));
+                DatabaseTopology.PartOfCluster = state != RachisConsensus.State.Passive;
+                if (DatabaseTopology.PartOfCluster == false)
```

Comment:
Wrong, can't handle this globally.
During tests we have multiple servers, with some in cluster and some not.